### PR TITLE
chore(deps): bump go to 1.20

### PR DESCRIPTION
**Pull request by mochaaP**
Review needed!

- switch to upstream quic-go
- drop go 1.18 as quic-go no longer supports it
- update import paths

### CODE REVIEW

**Comment by mochaaP** [comment]
(fix #888)

**Comment by mochaaP** [comment]
@DevinCarr Hello Devin, thank you for your time. Would you kindly review this pull request? I appreciate your help. :)

File: go.mod [hunk 1]
```diff
@@ -19,14 +19,14 @@ require (
...
 	github.com/json-iterator/go v1.1.12
-	github.com/lucas-clemente/quic-go v0.28.1
 	github.com/mattn/go-colorable v0.1.13
 	github.com/miekg/dns v1.1.50
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.13.0
 	github.com/prometheus/client_model v0.2.0
-	github.com/rs/zerolog v1.20.0
+	github.com/quic-go/quic-go v0.33.0
```
Comment (bompus) [inline go.mod:28]: can quic-go be bumped to v0.34.0 ? looks like it has a decent amount of fixes compared to 0.33.0

**Comment by mochaaP** [comment]
cc @bwesterb

**Comment by bwesterb** [comment]
lgtm. Didn't check `vendor` changes are correct.

**Comment by DevinCarr** [comment]
As I have mentioned before in the PR, we can't accept these changes without the following commit either applied to a new fork of quic-go or ideally merged upstream to quic-go: [Make max datagram frame size configurable: chungthuang/quic-go/commit/40b85c87](https://github.com/chungthuang/quic-go/commit/40b85c8763230e086f62ac310990dad55e4efa87)

We have not had the time yet to tackle this change since there was some API surface changes happening in quic-go that don't make the application of the commit just a rebase like it was previously.

**Comment by mochaaP** [comment]
@DevinCarr

I don't think that's necessary anymore. It was meant to fix the problem at <https://github.com/lucas-clemente/quic-go/issues/3273>, and it's resolved upsteam: <https://github.com/quic-go/quic-go/pull/3276>. What do you think?

**Comment by DevinCarr** [comment]
It was not meant to address the Windows MTU discoverability issue, that change was needed to set the max datagram size as defined in https://www.rfc-editor.org/rfc/rfc9221.html. It looks like there have been some changes in quic-go that make it easier to coordinate this change, but we still require this change to set the MaxDatagramFrameSize to 1350: https://github.com/cloudflare/cloudflared/blob/master/quic/param_unix.go#L6

I have started with a baseline of changes to add this to quic-go, but I haven't had the time yet to test it completely: https://github.com/DevinCarr/quic-go/commit/d1f4edacbee707b06bc5eeb0c33d90b12ce7fecc. Once this change is added to either a fork or merged upstream, we can evaluate the quic-go upgrade and go 1.20 changes to cloudflared.

**Comment by DevinCarr** [comment]
I have merged in the quic-go changes (https://github.com/cloudflare/cloudflared/commit/9426b603082905d0af8a07bdac866bc1d9c37cba) that should unblock the rest of this PR if you wish to restart it with just upgrading to go 1.20. Otherwise, I can close this and do it in the next couple of days.

**Comment by mochaaP** [comment]
@DevinCarr Hi Devin, I've made the changes as per your suggestions. Appreciate your efforts in addressing this upstream!

**Comment by 1715173329** [comment]
Well, hope this won't take forever to get merged.

**Review by 01SMD — APPROVED** [review]

**Review by 01SMD — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,107 +1,105 @@
...
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.13.0
-	github.com/prometheus/client_model v0.2.0
-	github.com/quic-go/quic-go v0.0.0-00010101000000-000000000000
-	github.com/rs/zerolog v1.20.0
-	github.com/stretchr/testify v1.8.1
-	github.com/urfave/cli/v2 v2.3.0
+	github.com/prometheus/client_golang v1.15.1
+	github.com/prometheus/client_model v0.4.0
+	github.com/quic-go/quic-go v0.34.0
```
Comment (stefanb) [inline go.mod:29]: To support go 1.20 and upcoming go 1.21 (currently at rc4) bump to 0.37.1:
```suggestion
	github.com/quic-go/quic-go v0.37.1
```

Comment (mochaaP) [reply go.mod:29]: waiting <https://github.com/DevinCarr/quic-go> to update/upstream

**Comment by joliveirinha** [comment]
We are going to address this internally because it requires some more careful details due to QUIC-GO. 
As of now we already compile with go1.20, and the next step will be 1.21 which as stated will require a little more effort.

For now, we will close this.

**Review by 01SMD — COMMENTED** [review]
.

**Review by 01SMD — COMMENTED** [review]
.

**Pull request closed without merging**